### PR TITLE
Fix: proxy command not run for initialization commands

### PR DESCRIPTION
**Pull request by Dhul-Husni**
## Why are these changes needed?
We use a proxy, [gcloud iap tunnel](https://cloud.google.com/iap/docs/using-tcp-forwarding), to access ray clusters with private ips. SSH proxy command works well for everything except "initialization_commands" where proxy commands are not passed.

## Related issue number

These PR's have relevant context: https://github.com/ray-project/ray/pull/10014, https://github.com/ray-project/ray/pull/8833

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Dhul-Husni** [comment]
@ericl @DmitriGekhtman @wuisawesome , Please review, this is a blocker for us. Thank you!

File: python/ray/autoscaler/_private/command_runner.py [hunk 1]
```diff
@@ -318,7 +318,7 @@ def run(
         if shutdown_after_run:
             cmd += "; sudo shutdown -h now"
         if ssh_options_override_ssh_key:
-            ssh_options = SSHOptions(ssh_options_override_ssh_key)
+            ssh_options = SSHOptions(ssh_options_override_ssh_key, ProxyCommand=self.ssh_proxy_command)
```
Comment (DmitriGekhtman) [inline python/ray/autoscaler/_private/command_runner.py:321]: I'm not sure what the consequence would be if self.ssh_proxy_command is the default `None`.
To de-risk this, would you mind writing this as follows: 
```suggestion
            if self.ssh_proxy_command:
                ssh_options = SSHOptions(ssh_options_override_ssh_key, ProxyCommand=self.ssh_proxy_command)
            else:
                ssh_options = SSHOptions(ssh_options_override_ssh_key)
```

Comment (DmitriGekhtman) [reply python/ray/autoscaler/_private/command_runner.py:321]: I don't know a lot about this code path.

The git blame on this points to a PR of @AmeerHajAli reviewed by @ericl and @ijrsvt in August 2020.
@AmeerHajAli @ericl @ijrsvt any thoughts?

Comment (ericl) [reply python/ray/autoscaler/_private/command_runner.py:321]: I think it ends up passed directly to SSH as ProxyCommand=None, which is probably bad. We should avoid setting it if None.

Comment (Dhul-Husni) [reply python/ray/autoscaler/_private/command_runner.py]: Thanks @DmitriGekhtman and @ericl I've updated the PR. It looks like Proxycommand does not get passed to ssh if it is None, see: https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/command_runner.py#L188-L192, still, to be careful, I updated the PR like you suggested.

**Comment by DmitriGekhtman** [comment]
@Dhul-Husni could you confirm that the changes in this PR work as expected for you? (We don't have per-commit test coverage for this code.)

**Review by ericl — APPROVED** [review]

**Review by ijrsvt — APPROVED** [review]
This change looks good to me as well!

**Comment by DmitriGekhtman** [comment]
You can fix the lint issue by running scripts/format.sh from the root of your local Ray repo.

**Comment by Dhul-Husni** [comment]
> @Dhul-Husni could you confirm that the changes in this PR work as expected for you? (We don't have per-commit test coverage for this code.)

@DmitriGekhtman Testing... Will let you know soon.

**Review by DmitriGekhtman — APPROVED** [review]

**Pull request merged**